sklearn/decomposition/_dict_learning.py

```diff
@@ -1479,6 +1591,30 @@ class MiniBatchDictionaryLearning(_BaseSparseCoding, BaseEstimator):
 
         .. versionadded:: 0.22
 
+    callback : callable, default=None
+        callable that gets invoked at the end of each iteration.
+
+        .. versionadded:: 1.0
+
+    tol : float, default=1e-3
+        Control early stopping based on the norm of the differences in the
+        dictionary between 2 steps.
+
+        To disable early stopping based on changes in the dictionary, set
+        `tol` to 0.0.
+
+        .. versionadded:: 1.0
+
+    max_no_improvement : int, default=10
+        Control early stopping based on the consecutive number of mini batches
+        that does not yield an improvement on the smoothed cost function.
```

Comment:
I am wondering about consistency in the scikit-learn API for controlling early stopping. Others estimators have `n_iter_no_change` for full-epochs checks instead of per-stochastic-step convergence checks.

The latter gives more control to the user because if the dataset is really really large, it might be the case that we would like to declare convergence even before scanning all the datapoints, that is before the end of the first epoch. But maybe this is a YAGNI and using the end-of-epoch `n_iter_no_change` strategy is enough in practice and more consistent with other estimators.

I am not really decided.

Reply:
Ok for keeping it that way then.